<!DOCTYPE HTML>
<html lang="en" >
    
    <head>
        
        <meta charset="UTF-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <title>Working with FASTQ files | R for Bioinformatics</title>
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <meta name="description" content="">
        <meta name="generator" content="GitBook 2.1.0">
        
        
        <meta name="HandheldFriendly" content="true"/>
        <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
        <meta name="apple-mobile-web-app-capable" content="yes">
        <meta name="apple-mobile-web-app-status-bar-style" content="black">
        <link rel="apple-touch-icon-precomposed" sizes="152x152" href="../gitbook/images/apple-touch-icon-precomposed-152.png">
        <link rel="shortcut icon" href="../gitbook/images/favicon.ico" type="image/x-icon">
        
    <link rel="stylesheet" href="../gitbook/style.css">
    
    

        
    
    
    <link rel="next" href="../GLOSSARY.html" />
    
    
    <link rel="prev" href="../bio1/index.html" />
    

        
    </head>
    <body>
        
        
    <div class="book" data-level="6.1" data-basepath=".." data-revision="Mon Jul 06 2015 17:51:11 GMT-0400 (EDT)">
    

<div class="book-summary">
    <div class="book-search">
        <input type="text" placeholder="Type to search" class="form-control" />
    </div>
    <ul class="summary">
        
        
        
        

        

        
    
        <li class="chapter " data-level="0" data-path="index.html">
            
                
                    <a href="../index.html">
                        <i class="fa fa-check"></i>
                        
                        Introduction
                    </a>
                
            
            
        </li>
    
        <li class="chapter " data-level="1" data-path="section1/index.html">
            
                
                    <a href="../section1/index.html">
                        <i class="fa fa-check"></i>
                        
                            <b>1.</b>
                        
                        Overview of R
                    </a>
                
            
            
        </li>
    
        <li class="chapter " data-level="2" data-path="section2/index.html">
            
                
                    <a href="../section2/index.html">
                        <i class="fa fa-check"></i>
                        
                            <b>2.</b>
                        
                        Getting Started with R
                    </a>
                
            
            
            <ul class="articles">
                
    
        <li class="chapter " data-level="2.1" data-path="section2/Installation.html">
            
                
                    <a href="../section2/Installation.html">
                        <i class="fa fa-check"></i>
                        
                            <b>2.1.</b>
                        
                        Installation
                    </a>
                
            
            
        </li>
    
        <li class="chapter " data-level="2.2" data-path="section2/InteractingWithR.html">
            
                
                    <a href="../section2/InteractingWithR.html">
                        <i class="fa fa-check"></i>
                        
                            <b>2.2.</b>
                        
                        Interacting with the R command line
                    </a>
                
            
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="3" data-path="section3/index.html">
            
                
                    <a href="../section3/index.html">
                        <i class="fa fa-check"></i>
                        
                            <b>3.</b>
                        
                        Data Types
                    </a>
                
            
            
            <ul class="articles">
                
    
        <li class="chapter " data-level="3.1" data-path="section3/vectors.html">
            
                
                    <a href="../section3/vectors.html">
                        <i class="fa fa-check"></i>
                        
                            <b>3.1.</b>
                        
                        Vectors
                    </a>
                
            
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="4" data-path="graphics/index.html">
            
                
                    <a href="../graphics/index.html">
                        <i class="fa fa-check"></i>
                        
                            <b>4.</b>
                        
                        Graphics
                    </a>
                
            
            
            <ul class="articles">
                
    
        <li class="chapter " data-level="4.1" data-path="graphics/ggplot2.html">
            
                
                    <a href="../graphics/ggplot2.html">
                        <i class="fa fa-check"></i>
                        
                            <b>4.1.</b>
                        
                        ggplot2
                    </a>
                
            
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="5" data-path="exercises/index.html">
            
                
                    <a href="../exercises/index.html">
                        <i class="fa fa-check"></i>
                        
                            <b>5.</b>
                        
                        Exercises and Workflows
                    </a>
                
            
            
            <ul class="articles">
                
    
        <li class="chapter " data-level="5.1" data-path="exercises/Data-exercises.html">
            
                
                    <a href="../exercises/Data-exercises.html">
                        <i class="fa fa-check"></i>
                        
                            <b>5.1.</b>
                        
                        Data exploration exercises
                    </a>
                
            
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="6" data-path="bio1/index.html">
            
                
                    <a href="../bio1/index.html">
                        <i class="fa fa-check"></i>
                        
                            <b>6.</b>
                        
                        Biological Applications
                    </a>
                
            
            
            <ul class="articles">
                
    
        <li class="chapter active" data-level="6.1" data-path="bio1/FASTQ.html">
            
                
                    <a href="../bio1/FASTQ.html">
                        <i class="fa fa-check"></i>
                        
                            <b>6.1.</b>
                        
                        Working with FASTQ files
                    </a>
                
            
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="7" data-path="GLOSSARY.html">
            
                
                    <a href="../GLOSSARY.html">
                        <i class="fa fa-check"></i>
                        
                            <b>7.</b>
                        
                        Glossary
                    </a>
                
            
            
        </li>
    


        
        <li class="divider"></li>
        <li>
            <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
                Published with GitBook
            </a>
        </li>
        
    </ul>
</div>

    <div class="book-body">
        <div class="body-inner">
            <div class="book-header">
    <!-- Actions Left -->
    <a href="#" class="btn pull-left toggle-summary" aria-label="Table of Contents"><i class="fa fa-align-justify"></i></a>
    <a href="#" class="btn pull-left toggle-search" aria-label="Search"><i class="fa fa-search"></i></a>
    
    <a href="../GLOSSARY.html" class="btn pull-left" aria-label="Glossary"><i class="fa fa-sort-alpha-asc"></i></a>
    
    <div id="font-settings-wrapper" class="dropdown pull-left">
        <a href="#" class="btn toggle-dropdown" aria-label="Font Settings"><i class="fa fa-font"></i>
        </a>
        <div class="dropdown-menu font-settings">
    <div class="dropdown-caret">
        <span class="caret-outer"></span>
        <span class="caret-inner"></span>
    </div>

    <div class="buttons">
        <button type="button" id="reduce-font-size" class="button size-2">A</button>
        <button type="button" id="enlarge-font-size" class="button size-2">A</button>
    </div>

    <div class="buttons font-family-list">
        <button type="button" data-font="0" class="button">Serif</button>
        <button type="button" data-font="1" class="button">Sans</button>
    </div>

    <div class="buttons color-theme-list">
        <button type="button" id="color-theme-preview-0" class="button size-3" data-theme="0">White</button>
        <button type="button" id="color-theme-preview-1" class="button size-3" data-theme="1">Sepia</button>
        <button type="button" id="color-theme-preview-2" class="button size-3" data-theme="2">Night</button>
    </div>
</div>

    </div>

    <!-- Actions Right -->
    
    <div class="dropdown pull-right">
        <a href="#" class="btn toggle-dropdown" aria-label="Share"><i class="fa fa-share-alt"></i>
        </a>
        <div class="dropdown-menu font-settings dropdown-left">
            <div class="dropdown-caret">
                <span class="caret-outer"></span>
                <span class="caret-inner"></span>
            </div>
            <div class="buttons">
                <button type="button" data-sharing="twitter" class="button">
                    Share on Twitter
                </button>
                <button type="button" data-sharing="google-plus" class="button">
                    Share on Google
                </button>
                <button type="button" data-sharing="facebook" class="button">
                    Share on Facebook
                </button>
                <button type="button" data-sharing="weibo" class="button">
                    Share on Weibo
                </button>
                <button type="button" data-sharing="instapaper" class="button">
                    Share on Instapaper
                </button>
            </div>
        </div>
    </div>
    

    
    <a href="#" target="_blank" class="btn pull-right google-plus-sharing-link sharing-link" data-sharing="google-plus" aria-label="Google"><i class="fa fa-google-plus"></i></a>
    
    
    <a href="#" target="_blank" class="btn pull-right facebook-sharing-link sharing-link" data-sharing="facebook" aria-label="Facebook"><i class="fa fa-facebook"></i></a>
    
    
    <a href="#" target="_blank" class="btn pull-right twitter-sharing-link sharing-link" data-sharing="twitter" aria-label="Twitter"><i class="fa fa-twitter"></i></a>
    
    
    


    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href="../" >R for Bioinformatics</a>
    </h1>
</div>

            <div class="page-wrapper" tabindex="-1">
                <div class="page-inner">
                
                
                    <section class="normal" id="section-">
                    
                        <h2 id="background">Background</h2>
<p>The <a href="../GLOSSARY.html#fastq" class="glossary-term" title="[FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format) is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. Both the sequence letter and quality score are each encoded with a single ASCII character for brevity. It was originally developed at the Wellcome Trust Sanger Institute to bundle a FASTA sequence and its quality data, but has recently become the de facto standard for storing the output of high-throughput sequencing instruments such as the Illumina Genome Analyzer.">FASTQ</a> format is a standard for storing sequence data and associated quality scores in a simple 
text format file. A first step in many analyses is to perform basic quality control on the <a href="../GLOSSARY.html#fastq" class="glossary-term" title="[FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format) is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. Both the sequence letter and quality score are each encoded with a single ASCII character for brevity. It was originally developed at the Wellcome Trust Sanger Institute to bundle a FASTA sequence and its quality data, but has recently become the de facto standard for storing the output of high-throughput sequencing instruments such as the Illumina Genome Analyzer.">FASTQ</a>
files. The Bioconductor ShortRead package is quite useful for doing this with very little code.</p>
<h2 id="load-libraries">Load libraries</h2>
<p>First, if the ShortRead package has not been installed, install it using the standard Biodonductor
installation process.</p>
<pre><code class="lang-r"><span class="hljs-keyword">source</span>(<span class="hljs-string">&apos;http://bioconductor.org/biocLite.R&apos;</span>)
biocLite(<span class="hljs-string">&apos;ShortRead&apos;</span>)
</code></pre>
<p>Once installed, load the library:</p>
<pre><code class="lang-r"><span class="hljs-keyword">library</span>(ShortRead)
</code></pre>
<h2 id="fastq-quality-control"><a href="../GLOSSARY.html#fastq" class="glossary-term" title="[FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format) is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. Both the sequence letter and quality score are each encoded with a single ASCII character for brevity. It was originally developed at the Wellcome Trust Sanger Institute to bundle a FASTA sequence and its quality data, but has recently become the de facto standard for storing the output of high-throughput sequencing instruments such as the Illumina Genome Analyzer.">FASTQ</a> Quality Control</h2>
<h3 id="quality-by-cycle">Quality-by-cycle</h3>
<p>The ShortRead package includes some example <a href="../GLOSSARY.html#fastq" class="glossary-term" title="[FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format) is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. Both the sequence letter and quality score are each encoded with a single ASCII character for brevity. It was originally developed at the Wellcome Trust Sanger Institute to bundle a FASTA sequence and its quality data, but has recently become the de facto standard for storing the output of high-throughput sequencing instruments such as the Illumina Genome Analyzer.">FASTQ</a> files. The way that we can find those files is
to use <code>system.file</code>.  This function looks up the location of the ShortRead installation and 
then finds files relative to that location. In this case, we are going to get the file locations
for two small <a href="../GLOSSARY.html#fastq" class="glossary-term" title="[FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format) is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. Both the sequence letter and quality score are each encoded with a single ASCII character for brevity. It was originally developed at the Wellcome Trust Sanger Institute to bundle a FASTA sequence and its quality data, but has recently become the de facto standard for storing the output of high-throughput sequencing instruments such as the Illumina Genome Analyzer.">FASTQ</a> files.</p>
<pre><code class="lang-r"><span class="hljs-comment"># I just know from previous looking that the <a href="../GLOSSARY.html#fastq" class="glossary-term" title="[FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format) is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. Both the sequence letter and quality score are each encoded with a single ASCII character for brevity. It was originally developed at the Wellcome Trust Sanger Institute to bundle a FASTA sequence and its quality data, but has recently become the de facto standard for storing the output of high-throughput sequencing instruments such as the Illumina Genome Analyzer.">fastq</a> files are here</span>
fastqDir = system.file(package=<span class="hljs-string">&apos;ShortRead&apos;</span>,<span class="hljs-string">&apos;extdata/E-MTAB-1147&apos;</span>)
</code></pre>
<p>Now, we can read the <a href="../GLOSSARY.html#fastq" class="glossary-term" title="[FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format) is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. Both the sequence letter and quality score are each encoded with a single ASCII character for brevity. It was originally developed at the Wellcome Trust Sanger Institute to bundle a FASTA sequence and its quality data, but has recently become the de facto standard for storing the output of high-throughput sequencing instruments such as the Illumina Genome Analyzer.">FASTQ</a> files in this directory using <code>readFastq</code>.</p>
<pre><code class="lang-r">fq = readFastq(dirPath = fastqDir,pattern=<span class="hljs-string">&apos;*.<a href="../GLOSSARY.html#fastq" class="glossary-term" title="[FASTQ format](https://en.wikipedia.org/wiki/FASTQ_format) is a text-based format for storing both a biological sequence (usually nucleotide sequence) and its corresponding quality scores. Both the sequence letter and quality score are each encoded with a single ASCII character for brevity. It was originally developed at the Wellcome Trust Sanger Institute to bundle a FASTA sequence and its quality data, but has recently become the de facto standard for storing the output of high-throughput sequencing instruments such as the Illumina Genome Analyzer.">fastq</a>.gz&apos;</span>)
</code></pre>
<pre><code>## Warning: closing unused connection 69
## (/Users/sdavis2/Documents/git/RForBioinformatics/section3/vectors.Rmd)
</code></pre><pre><code>## Warning: closing unused connection 68
## (/Users/sdavis2/Documents/git/RForBioinformatics/section3/vectors.md)
</code></pre><pre><code>## Warning: closing unused connection 67
## (/Users/sdavis2/Documents/git/RForBioinformatics/section2/InteractingWithR.Rmd)
</code></pre><pre><code>## Warning: closing unused connection 66
## (/Users/sdavis2/Documents/git/RForBioinformatics/section2/InteractingWithR.md)
</code></pre><pre><code>## Warning: closing unused connection 65
## (/Users/sdavis2/Documents/git/RForBioinformatics/section2/Installation.mmd)
</code></pre><pre><code>## Warning: closing unused connection 64
## (/Users/sdavis2/Documents/git/RForBioinformatics/section2/Installation.md)
</code></pre><pre><code>## Warning: closing unused connection 63
## (/Users/sdavis2/Documents/git/RForBioinformatics/section1/README.Rmd)
</code></pre><pre><code>## Warning: closing unused connection 62
## (/Users/sdavis2/Documents/git/RForBioinformatics/references.Rmd)
</code></pre><pre><code>## Warning: closing unused connection 61
## (/Users/sdavis2/Documents/git/RForBioinformatics/references.md)
</code></pre><pre><code>## Warning: closing unused connection 60
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/yargs/node_modules/decamelize/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 59
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/yargs/node_modules/camelcase/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 58
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/yargs/CHANGELOG.md)
</code></pre><pre><code>## Warning: closing unused connection 57
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/speech-rule-engine/node_modules/xmldom/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 56
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/speech-rule-engine/node_modules/commander/Readme.md)
</code></pre><pre><code>## Warning: closing unused connection 55
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/speech-rule-engine/node_modules/commander/History.md)
</code></pre><pre><code>## Warning: closing unused connection 54
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/qs/CONTRIBUTING.md)
</code></pre><pre><code>## Warning: closing unused connection 53
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/qs/CHANGELOG.md)
</code></pre><pre><code>## Warning: closing unused connection 52
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/node-uuid/LICENSE.md)
</code></pre><pre><code>## Warning: closing unused connection 51
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/mime-types/node_modules/mime-db/HISTORY.md)
</code></pre><pre><code>## Warning: closing unused connection 50
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/mime-types/HISTORY.md)
</code></pre><pre><code>## Warning: closing unused connection 49
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/json-stringify-safe/CHANGELOG.md)
</code></pre><pre><code>## Warning: closing unused connection 48
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/isstream/LICENSE.md)
</code></pre><pre><code>## Warning: closing unused connection 47
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/http-signature/http_signing.md)
</code></pre><pre><code>## Warning: closing unused connection 46
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/hawk/node_modules/hoek/CONTRIBUTING.md)
</code></pre><pre><code>## Warning: closing unused connection 45
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/hawk/node_modules/boom/CONTRIBUTING.md)
</code></pre><pre><code>## Warning: closing unused connection 44
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/commander/Readme.md)
</code></pre><pre><code>## Warning: closing unused connection 43
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/commander/History.md)
</code></pre><pre><code>## Warning: closing unused connection 42
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/chalk/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 41
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/chalk/node_modules/supports-color/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 40
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/chalk/node_modules/strip-ansi/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 39
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/chalk/node_modules/strip-ansi/node_modules/ansi-regex/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 38
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/chalk/node_modules/has-ansi/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 37
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/chalk/node_modules/has-ansi/node_modules/ansi-regex/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 36
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/chalk/node_modules/escape-string-regexp/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 35
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/har-validator/node_modules/chalk/node_modules/ansi-styles/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 34
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/form-data/Readme.md)
</code></pre><pre><code>## Warning: closing unused connection 33
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/form-data/node_modules/mime-types/node_modules/mime-db/HISTORY.md)
</code></pre><pre><code>## Warning: closing unused connection 32
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/form-data/node_modules/mime-types/HISTORY.md)
</code></pre><pre><code>## Warning: closing unused connection 31
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/form-data/node_modules/async/CHANGELOG.md)
</code></pre><pre><code>## Warning: closing unused connection 30
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/extend/CHANGELOG.md)
</code></pre><pre><code>## Warning: closing unused connection 29
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/combined-stream/Readme.md)
</code></pre><pre><code>## Warning: closing unused connection 28
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/combined-stream/node_modules/delayed-stream/Readme.md)
</code></pre><pre><code>## Warning: closing unused connection 27
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/node_modules/bl/LICENSE.md)
</code></pre><pre><code>## Warning: closing unused connection 26
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/CONTRIBUTING.md)
</code></pre><pre><code>## Warning: closing unused connection 25
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/request/CHANGELOG.md)
</code></pre><pre><code>## Warning: closing unused connection 24
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/parse5/CHANGELOG.md)
</code></pre><pre><code>## Warning: closing unused connection 23
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/htmlparser2/node_modules/entities/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 22
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/htmlparser2/node_modules/domutils/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 21
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/htmlparser2/node_modules/domutils/node_modules/dom-serializer/node_modules/entities/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 20
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/htmlparser2/node_modules/domutils/node_modules/dom-serializer/node_modules/domelementtype/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 19
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/htmlparser2/node_modules/domhandler/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 18
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/htmlparser2/node_modules/domelementtype/readme.md)
</code></pre><pre><code>## Warning: closing unused connection 17
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/cssom/README.mdown)
</code></pre><pre><code>## Warning: closing unused connection 16
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/contextify/node_modules/nan/LICENSE.md)
</code></pre><pre><code>## Warning: closing unused connection 15
## (/Users/sdavis2/Documents/git/RForBioinformatics/node_modules/gitbook-plugin-mathjax/node_modules/MathJax-node/node_modules/jsdom/node_modules/contextify/node_modules/nan/CHANGELOG.md)
</code></pre><pre><code>## Warning: closing unused connection 14
## (/Users/sdavis2/Documents/git/RForBioinformatics/InteractingWithR.md)
</code></pre><pre><code>## Warning: closing unused connection 13
## (/Users/sdavis2/Documents/git/RForBioinformatics/graphics/ggplot2.Rmd)
</code></pre><pre><code>## Warning: closing unused connection 12
## (/Users/sdavis2/Documents/git/RForBioinformatics/graphics/ggplot2.md)
</code></pre><pre><code>## Warning: closing unused connection 11
## (/Users/sdavis2/Documents/git/RForBioinformatics/GLOSSARY.md)
</code></pre><pre><code>## Warning: closing unused connection 10
## (/Users/sdavis2/Documents/git/RForBioinformatics/exercises/Data-exercises.Rmd)
</code></pre><pre><code>## Warning: closing unused connection 9
## (/Users/sdavis2/Documents/git/RForBioinformatics/exercises/Data-exercises.md)
</code></pre><pre><code>## Warning: closing unused connection 8
## (/Users/sdavis2/Documents/git/RForBioinformatics/bio1/FASTQ.Rmd)
</code></pre><pre><code>## Warning: closing unused connection 7
## (/Users/sdavis2/Documents/git/RForBioinformatics/bio1/FASTQ.md)
</code></pre><pre><code>## Warning: closing unused connection 6
## (/Users/sdavis2/Documents/git/RForBioinformatics/_book/section2/InteractingWithR.md)
</code></pre><pre><code>## Warning: closing unused connection 5
## (/Users/sdavis2/Documents/git/RForBioinformatics/_book/section2/Installation.mmd)
</code></pre><pre><code>## Warning: closing unused connection 4
## (/Users/sdavis2/Documents/git/RForBioinformatics/_book/references.md)
</code></pre><pre><code class="lang-r">fq
</code></pre>
<pre><code>## class: ShortReadQ
## length: 40000 reads; width: 72 cycles
</code></pre><p>The <code>fq</code> object now contains 40000 reads, each 72 bp (cycles) long. Next, we are going to 
calculate the number of bases with each quality over each cycle.  </p>
<pre><code class="lang-r">m = alphabetByCycle(quality(fq))
dim(m)
</code></pre>
<pre><code>## [1] 94 72
</code></pre><pre><code class="lang-r">head(m[,<span class="hljs-number">1</span>:<span class="hljs-number">8</span>])
</code></pre>
<pre><code>##         cycle
## alphabet [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8]
##             0    0    0    0    0    0    0    0
##        !    0    0    0    0    0    0    0    0
##        &quot;    0    0    0    0    0    0    0    0
##        #  199  220  233  241  249  274  289  304
##        $    0    0    0    1    2    0    0    0
##        %    2    2    6    0   89    7    0   11
</code></pre><p>So, we have a matrix of numbers and we&apos;d like to see if there is any structure in those numbers. A heatmap can be very useful for looking at matrices, so we can try that here.</p>
<pre><code class="lang-r"><span class="hljs-keyword">library</span>(gplots)
heatmap.2(log10(m[<span class="hljs-number">4</span>:<span class="hljs-number">40</span>,]+<span class="hljs-number">1</span>),Rowv = <span class="hljs-literal">NA</span>, Colv=<span class="hljs-literal">NA</span>,trace=<span class="hljs-string">&quot;none&quot;</span>)
</code></pre>
<pre><code>## Warning in heatmap.2(log10(m[4:40, ] + 1), Rowv = NA, Colv = NA, trace =
## &quot;none&quot;): Discrepancy: Rowv is FALSE, while dendrogram is `none&apos;. Omitting
## row dendogram.
</code></pre><p><img src="figure/heatmap-1.png" alt="plot of chunk heatmap"> </p>
<ul>
<li>What does the heatmap show?</li>
</ul>
<h3 id="nucleotide-by-cycle-plot">Nucleotide-by-cycle plot</h3>
<p>We can also look a the base-by-cycle of the actual sequences in our data using a similar approach.
\</p>
<pre><code class="lang-r">m = alphabetByCycle(sread(fq))
head(m[,<span class="hljs-number">1</span>:<span class="hljs-number">8</span>])
</code></pre>
<pre><code>##         cycle
## alphabet  [,1]  [,2]  [,3]  [,4]  [,5]  [,6]  [,7]  [,8]
##        A  4253  9782  9965 11241 11512 13574  5237  7621
##        C 21264  8913  9913  9533  8108 10664 10524 12942
##        G  5193 11617 12729 13279 13345 10337  8770 11097
##        T  9267  9677  7376  5937  6862  5413 15461  8331
##        M     0     0     0     0     0     0     0     0
##        R     0     0     0     0     0     0     0     0
</code></pre><p>We can pull out only the bases we are interested in and then make a matrix plot to look at the 
base biases along the length of the reads:</p>
<pre><code class="lang-r">ms = t(m[c(<span class="hljs-string">&apos;A&apos;</span>,<span class="hljs-string">&apos;C&apos;</span>,<span class="hljs-string">&apos;G&apos;</span>,<span class="hljs-string">&apos;T&apos;</span>,<span class="hljs-string">&apos;N&apos;</span>),])
matplot(ms,type=<span class="hljs-string">&apos;l&apos;</span>,xlab=<span class="hljs-string">&apos;cycle&apos;</span>,ylab=<span class="hljs-string">&apos;count&apos;</span>,main=<span class="hljs-string">&apos;Alphabet by Cycle&apos;</span>)
</code></pre>
<p><img src="figure/matplot-1.png" alt="plot of chunk matplot"> </p>
<h2 id="session-info">Session Info</h2>
<pre><code class="lang-r">sessionInfo()
</code></pre>
<pre><code>## R version 3.2.1 (2015-06-18)
## Platform: x86_64-apple-darwin13.4.0 (64-bit)
## Running under: OS X 10.10 (Yosemite)
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] biomaRt_2.23.5           randomForest_4.6-10     
##  [3] gplots_2.16.0            ShortRead_1.25.8        
##  [5] GenomicAlignments_1.3.28 Rsamtools_1.19.35       
##  [7] GenomicRanges_1.20.3     GenomeInfoDb_1.3.13     
##  [9] Biostrings_2.35.11       XVector_0.7.4           
## [11] IRanges_2.2.1            S4Vectors_0.6.0         
## [13] BiocParallel_1.1.13      BiocGenerics_0.14.0     
## [15] nycflights13_0.1         ggplot2_1.0.1           
## [17] BiocStyle_1.5.3          knitr_1.8               
## [19] Rgitbook_0.9            
## 
## loaded via a namespace (and not attached):
##  [1] gtools_3.4.1         reshape2_1.4         knitcitations_1.0.6 
##  [4] lattice_0.20-31      colorspace_1.2-4     base64enc_0.1-2     
##  [7] XML_3.98-1.1         DBI_0.3.1            RColorBrewer_1.0-5  
## [10] foreach_1.4.2        plyr_1.8.1           stringr_0.6.2       
## [13] zlibbioc_1.13.1      munsell_0.4.2        gtable_0.1.2        
## [16] hwriter_1.3.2        caTools_1.17.1       codetools_0.2-11    
## [19] memoise_0.2.1        evaluate_0.5.5       labeling_0.3        
## [22] latticeExtra_0.6-26  Biobase_2.27.1       AnnotationDbi_1.30.1
## [25] proto_0.3-10         Rcpp_0.11.6          KernSmooth_2.23-14  
## [28] scales_0.2.4         checkmate_1.5.0      formatR_1.0         
## [31] gdata_2.13.3         sendmailR_1.2-1      brew_1.0-6          
## [34] BatchJobs_1.5        fail_1.2             digest_0.6.8        
## [37] BBmisc_1.8           RJSONIO_1.3-0        grid_3.2.1          
## [40] bibtex_0.4.0         tools_3.2.1          bitops_1.0-6        
## [43] RCurl_1.95-4.3       RSQLite_1.0.0        RefManageR_0.8.63   
## [46] MASS_7.3-40          lubridate_1.3.3      httr_0.6.1          
## [49] iterators_1.0.7
</code></pre>
                    
                    </section>
                
                
                </div>
            </div>
        </div>

        
        <a href="../bio1/index.html" class="navigation navigation-prev " aria-label="Previous page: Biological Applications"><i class="fa fa-angle-left"></i></a>
        
        
        <a href="../GLOSSARY.html" class="navigation navigation-next " aria-label="Next page: Glossary"><i class="fa fa-angle-right"></i></a>
        
    </div>
</div>

        
<script src="../gitbook/app.js"></script>

    
    <script src="https://cdn.mathjax.org/mathjax/2.5-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    

    
    <script src="../gitbook/plugins/gitbook-plugin-mathjax/plugin.js"></script>
    

    
    <script src="../gitbook/plugins/gitbook-plugin-ga/plugin.js"></script>
    

    
    <script src="../gitbook/plugins/gitbook-plugin-livereload/plugin.js"></script>
    

<script>
require(["gitbook"], function(gitbook) {
    var config = {"fontSettings":{"theme":null,"family":"sans","size":2},"ga":{"token":"UA-2851481-1","configuration":"auto"}};
    gitbook.start(config);
});
</script>

        <script>(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,'script','//www.google-analytics.com/analytics.js','ga');ga('create', 'UA-2851481-1', 'auto');ga('send', 'pageview');</script>
    </body>
    
</html>
